src/coreclr/pal/src/thread/context.cpp

```diff
@@ -1717,6 +1717,12 @@ CONTEXT_GetThreadContextFromThreadState(
             CONTEXT_GetThreadContextFromThreadState((thread_state_flavor_t)pState->fsh.flavor, (thread_state_t)&pState->ufs, lpContext);
         }
         break;
+    case x86_AVX_STATE:
```

Comment:
Does this need to handle `x86_AVX512_STATE` as well?

Reply:
Is this really needed? I have thought that the x86_AVX_STATE64 / x86_AVX512_STATE64 would be the only ones that we would get on 64 bit CPUs. The x86_avx_state_t is a union containing x86_avx_state64_t or x86_avx_state32_t and we don't support 32 bit x86 on macos.

Reply:
On line 1294 of machexception.cpp, there is a call to `thread_get_state`. I tried calling `thread_get_state` with the second parameter as x86_AVX_STATE64, but that returned an invalid argument error. When I changed the second parameter to x86_AVX_STATE, `thread_get_state` returned with no errors. 

This is why I followed the convention of the other thread states and recursively called `Context_GetThreadContextFromThreadState`. At this point, the thread state will be of type: x86_AVX_STATEz.

Reply:
Ah, right, I've forgotten that the OS returns the xx_STATE and we always need to pick the 64 bit one from it (that's what the recursion does). So we need to add case for both `x86_AVX_STATE` and `x86_AVX512_STATE` here